test/test_suite/suite_filter.test.ts

```diff
@@ -0,0 +1,96 @@
+import { assert } from 'chai';
+import 'mocha';
+
+import { suiteFilter } from '../../src/test_suite';
+
+describe('suite_filter', () => {
+    ///////////////////////////////////////////////////////////////////////////////
+    //
+    //  Suite Filter
+    //
+    ///////////////////////////////////////////////////////////////////////////////
+    const suites = ['a', 'b', 'c', 'd'];
+    const suites2 = ['foo', 'bar', 'foo-bar'];
+
+    it('parse', () => {
+        // Simple TERM
```

Comment:
I feel like all of these headings should be separated into different unit tests so that it's more clear on what functionality failed. I would also prefer to see more descriptive test names e.g.

`it('should parse blah blah....', () =>` 

I usually try to avoid having one unit test with a lot of assert statements in it; would like to hear your thoughts on this!